### PR TITLE
Refactor users store

### PR DESCRIPTION
**Pull request by bzang**
Created new users module. Migrated recents module over to it. Need substantially more work to get the rest of the widgets over.

### CODE REVIEW

**Comment by adamweeks** [comment]
Deploy preview ready!

Built with commit 424af0fc6703f39232f797514a0fb881e70a1963

https://deploy-preview-613--hopeful-hamilton-4090b9.netlify.com

File: packages/node_modules/@ciscospark/redux-module-spaces/src/actions.js [hunk 1]
```diff
@@ -110,12 +114,15 @@ export function getSpaces(sparkInstance, options = {}) {
       items.forEach((space) => {
         if (!space.tags.includes(TAG_HIDDEN)) {
           spaces[space.id] = constructSpace(space, listOptions.deferDecrypt);
+          dispatch(storeUsers(space.participants.items));
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/redux-module-spaces/src/actions.js]: Should this go in an `else` after `if (space.decrypt)`?

File: packages/node_modules/@ciscospark/redux-module-users/src/actions.js [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    }
+  };
+}
+
+/**
+ * Retrieves the current user using internal APIs
+ * @param {Object} sparkInstance
+ * @returns {Function}
+ */
+export function fetchCurrentUser(sparkInstance) {
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/redux-module-users/src/actions.js]: Let's get some snapshot tests on this module.

File: packages/node_modules/@ciscospark/redux-module-spaces/src/actions.js (hunk 1 above)
Comment (bzang) [reply packages/node_modules/@ciscospark/redux-module-spaces/src/actions.js]: This is should happen even if we don't enable deferred decrypt. When we do defer, this line adds the encrypted users as placeholders. If we don't defer, this line will add the already decrypted users.

Comment (adamweeks) [reply packages/node_modules/@ciscospark/redux-module-spaces/src/actions.js]: Do we get a proper ID or email for encrypted users that will be updated after the decrypt?

**Comment by bzang** [comment]
@adamweeks , ready for round 2

File: packages/node_modules/@ciscospark/redux-module-spaces/src/actions.test.js [hunk 3]
```diff
@@ -153,12 +154,7 @@ describe('redux-module-space actions ', () => {
   });
 
   it('has exported actions', () => {
-    expect(actions.fetchSpaces).toBeDefined();
-    expect(actions.getSpaces).toBeDefined();
-    expect(actions.fetchSpace).toBeDefined();
-    expect(actions.removeSpace).toBeDefined();
-    expect(actions.updateSpaceWithActivity).toBeDefined();
-    expect(actions.updateSpaceRead).toBeDefined();
+    expect(actions).toMatchSnapshot();
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/redux-module-spaces/src/actions.test.js]: Oooh, I like that

**Pull request merged**